src/eventbridge/index.ts

```diff
@@ -0,0 +1,78 @@
+import { aws_events } from "aws-cdk-lib";
+import { Rule } from "aws-cdk-lib/aws-events";
+import { Construct } from "constructs";
+import { FunctionDecl } from "../declaration";
+import { synthesizeEventPattern } from "./eventpattern";
+import { synthesizeEventBridgeTarget } from "./targets";
+
+export interface EventBusEvent<
+  T = any,
+  DetailType extends string = string,
+  Source extends string = string
+> {
+  source: Source;
+  detailType: DetailType;
+  detail: T;
+}
+
+type EventHandlerFunction<E extends EventBusEvent = EventBusEvent<any>> = (
+  event: E
+) => void;
+
+export type EventPredicateFunction<
+  E extends EventBusEvent = EventBusEvent<any>
+> = (event: E) => boolean;
+
+export class EventBusTarget<T extends EventBusEvent> {
+  target: aws_events.IRuleTarget;
+
+  constructor(func: EventHandlerFunction<T>) {
+    const decl = func as unknown as FunctionDecl;
+    this.target = synthesizeEventBridgeTarget(decl);
+  }
+}
+
+export class EventBusRule<T extends EventBusEvent> extends Construct {
+  public readonly rule: aws_events.Rule;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    bus: aws_events.EventBus,
+    predicate: EventPredicateFunction<T>
+  ) {
+    super(scope, id);
+
+    const decl = predicate as unknown as FunctionDecl;
+    const pattern = synthesizeEventPattern(decl);
+
+    this.rule = new Rule(this, "rule", {
+      eventPattern: pattern as aws_events.EventPattern,
+      eventBus: bus,
+    });
+  }
+
+  target<E extends T>(target: EventHandlerFunction<E>): EventBusRule<T>;
+  target<E extends T>(target: EventBusTarget<E>): EventBusRule<T>;
+  target<E extends T>(
+    target: EventHandlerFunction<E> | EventBusTarget<E>
+  ): EventBusRule<T> {
+    const _target =
+      target instanceof EventBusTarget ? target : new EventBusTarget<E>(target);
+    this.rule.addTarget(_target.target);
+    return this;
+  }
+}
+
+export class EventBus<E extends EventBusEvent> {
```

Comment:
Let's make sure we thoroughly document these classes with high quality documentation, including examples. The docs here are what will be generated with `tsdoc` and included on the `functionless.org` website.